### PR TITLE
Support cache size of vineyard fuse and enable serverless mode for vineyard runtime

### PR DESCRIPTION
**Pull request by dashanji**
### Ⅰ. Describe what this PR does

- Support to configure the cache size of vineyard fuse.
- Enable the serverless mode for vineyard runtime.
- Support to configure the networkMode and podMetadata of vineyard components.
- Add the related docs.

### CODE REVIEW

File: pkg/common/constants.go [hunk 1]
```diff
@@ -187,6 +187,7 @@ const (
 	injectSidecar                 = ".sidecar" + inject
 	InjectServerless              = "serverless" + inject             // serverless.fluid.io/inject
 	InjectUnprivilegedFuseSidecar = "unprivileged" + injectSidecar    // unprivileged.sidecar.fluid.io/inject
+	InjectVineyardSidecar         = "vineyard" + injectSidecar        // vineyard.sidecar.fluid.io/inject
```
Comment (cheyang) [inline pkg/common/constants.go:190]: I think if it's no need  to set InjectVineyardSidecar by user. It should be determined automatically.

Comment (dashanji) [reply pkg/common/constants.go:190]: Okay, I get it.

**Comment by dashanji** [comment]
Hi @cheyang @TrafalgarZZZ , could you please take a look at this?

File: pkg/application/inject/fuse/injector.go [hunk 2]
```diff
@@ -289,3 +294,36 @@ func (s *Injector) labelInjectionDone(pod common.FluidObject) error {
 func (s *Injector) getServerlessPlatformFromMeta(metaObj metav1.ObjectMeta) string {
 	return utils.GetServerlessPlatfrom(metaObj.Labels)
 }
+
+// check whether the podSpecs is using vineyard runtime
+func (s *Injector) checkVineyardPlatform(podSpecs *mutator.MutatingPodSpecs, runtimeInfos map[string]base.RuntimeInfoInterface) string {
```
Comment (cheyang) [inline pkg/application/inject/fuse/injector.go:299]: I think Vineyard is a runtime not platform.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/application/inject/fuse/injector.go (hunk 2 above)
Comment (dashanji) [reply pkg/application/inject/fuse/injector.go:299]: Thanks for this catch.

**Pull request closed without merging**